decisions/0015-separate-library-per-product.md

```diff
@@ -0,0 +1,40 @@
+---
+parent: Decisions
+nav_order: 15
+adr: "0015"
+status: In progress
+date: 2022-11-03
+tags: [clients]
+---
+
+# Separate library per product
+
+## Context and Problem Statement
+
+Our `clients` repository currently stores much common logic inside of the `libs/common` package,
+which is a library that is consumed by all of our clients. However not all parts of the package are
+actually used by all of our clients which makes our bundles unecessarily large. Another issue with
+this arrangement is the increased cognitive load for the different pods that have to filter out the
+parts of the package that don't concern their work.
+
+## Considered Options
+
+- **Continue as is** - Place all services, models, views, etc in `libs/common`.
+- **Put services that are only used in 1 client inside that client** - This has the benefit of not
+  bloating the common package but will start "polluting" our frontends with request/response models
+  that we're trying to hide behind services. Cognitive load is also not particularly improved.
+- **Create new libs for each product** - This involves splitting our common code into `common`,
```

Comment:
How would this work with our need to use a separate Angular library to handle CLI?

Reply:
Can you elaborate a bit?